---
title: "Supplementary Info"
author: "James Ware"
date: "`r Sys.Date()`"
output:
  # html_document
  word_document:
    reference_docx: ./templateLandscape.docx
---

```{r setupNest, include=FALSE, cache=F}
knitr::opts_chunk$set(echo = FALSE)
```

##Supplementary File 3. HCM-causing variants cluster on residues involved in IHM-related inter- and intra-molecular interactions.

**_Table S3a - Pathogenic variants._** For each of the four main IHM interactions (IHM priming, anchoring, stabilizing and scaffolding), the number of distinct pathogenic variants (Table1, in the main text) affecting interacting residues (of a total of `r unique(outputPath$caseVarsTotal)` variants) is shown.  The length of the interaction site (no of amino acid residues; total protein length = 1935) is used to determine the proportion of variants that would be expected to lie in the region of interest under the null (a uniform distribution), and the rates are compared with a binomial test.

```{r S3a_forSupp, cache=FALSE}
knitr::kable(tableS3a,digits=50)
```

**_Table S3b - Likely pathogenic variants._** Shows equivalent data for `r unique(outputLP$caseVarsTotal)` likely pathogenic variants (Table 2, in the main text).  The clustering in the IHM priming and stabilizing regions is replicated.

```{r S3b_forSupp, cache=FALSE}
knitr::kable(tableS3b,digits=50)
```

**_Table S3c - All pathogenic / likely pathogenic variants._** Shows equivalent statistics when `r unique(outputPLP$caseVarsTotal)` pathogenic and likely pathogenic variants are analyzed in combination. This table appears in the main text as Table 4, duplicated here for convenience.


```{r S3c_forSupp, cache=FALSE}
knitr::kable(tableS3c,digits=50)
```

**_Table S3d - All interactions._** Here pathogenic and likely pathogenic variants are combined (total = `r unique(outputPandLP$caseVarsTotal)`), to empower interrogation of each individual interaction motif.  Interactions refer to definitions given in Supplementary File 2, and full details of all variants are shown in Tables 1, 2.

```{r S3d_forSupp, cache=FALSE}
knitr::kable(tableS3d,digits=50)
```

**_Table S3e - All MD functional sites._** Pathogenic and likely pathogenic variants are combined as above (total = `r unique(outputPandLP$caseVarsTotal)`), to empower interrogation of individual motor domain functional sites.  Full variant details are shown in Tables 1, 2.

```{r S3e_forSupp, cache=FALSE}
knitr::kable(tableS3e,digits=50)
```

**_Table S3f - Tabulating differential impact of variants located on the free head (FH) and blocked head (BH)._** Pathogenic and likely pathogenic variants are analyzed in combination.  Full variant details are shown in Tables 1, 2.

```{r S3f_forSupp, cache=FALSE}
knitr::kable(tableS3f,digits=50) 
```